src/Libraries/Microsoft.Extensions.Diagnostics.ResourceMonitoring/Windows/Disk/WindowsDiskIoTimePerfCounter.cs

```diff
@@ -0,0 +1,76 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+
+namespace Microsoft.Extensions.Diagnostics.ResourceMonitoring.Windows.Disk;
+
+internal sealed class WindowsDiskIoTimePerfCounter
+{
+    private readonly List<IPerformanceCounter> _counters = [];
+    private readonly IPerformanceCounterFactory _performanceCounterFactory;
+    private readonly TimeProvider _timeProvider;
+    private readonly string _categoryName;
+    private readonly string _counterName;
+    private readonly string[] _instanceNames;
+    private long _lastTimestamp;
+
+    internal WindowsDiskIoTimePerfCounter(
+        IPerformanceCounterFactory performanceCounterFactory,
+        TimeProvider timeProvider,
+        string categoryName,
+        string counterName,
+        string[] instanceNames)
+    {
+        _performanceCounterFactory = performanceCounterFactory;
+        _timeProvider = timeProvider;
+        _categoryName = categoryName;
+        _counterName = counterName;
+        _instanceNames = instanceNames;
+    }
+
+    /// <summary>
+    /// Gets the disk time measurements.
+    /// Key: Disk name, Value: Real elapsed time used in busy state.
+    /// </summary>
+    internal IDictionary<string, double> TotalSeconds { get; } = new ConcurrentDictionary<string, double>();
+
+    internal void InitializeDiskCounters()
+    {
+        foreach (string instanceName in _instanceNames)
+        {
+            // Create counters for each disk
+            _counters.Add(_performanceCounterFactory.Create(_categoryName, _counterName, instanceName));
+            TotalSeconds[instanceName] = 0f;
+        }
+
+        // Initialize the counters to get the first value
+        foreach (IPerformanceCounter counter in _counters)
+        {
+            _ = counter.NextValue();
+        }
+
+        _lastTimestamp = _timeProvider.GetUtcNow().ToUnixTimeMilliseconds();
+    }
+
+    internal void UpdateDiskCounters()
+    {
+        long currentTimestamp = _timeProvider.GetUtcNow().ToUnixTimeMilliseconds();
```

Comment:
```suggestion
        long currentTimestamp = _timeProvider.GetUtcNow().ToUnixTimeSeconds();
```

and avoid converting to seconds later because it is already seconds?

Reply:
I still prefer using `ToUnixTimeMilliseconds`, because if we use `ToUnixTimeSeconds`, the calculated time interval can only be accurate to the second. However, we cannot guarantee that the time difference between two samples is exactly an integer number of seconds, which may lead to precision issues.

Reply:
Ok, I see your point, thank you!

However, it has just occurred to me, why don't we use ticks? For instance, we do use ticks here https://github.com/dotnet/extensions/blob/9652f798f2de4e40c088075d00d185628c8e3dbb/src/Libraries/Microsoft.Extensions.Diagnostics.ResourceMonitoring/Windows/WindowsContainerSnapshotProvider.cs#L235. We can convert to seconds only at the very last moment before saving it to the `TotalSeconds` dictionary

Reply:
.